stl-decomp-4j/src/main/java/com/github/servicenow/ds/stats/stl/SeasonalTrendLoess.java

```diff
@@ -370,18 +376,44 @@ private void sanityCheck(double[] data) {
 							"SeasonalTrendLoess.Builder: setSeasonalWidth or setPeriodic must be called.");
 			}
 
-			if (fFlatTrend || fLinearTrend) {
-				if (fTrendWidth != null)
+			if (fFlatTrend) {
+
+				int massiveWidth = 100 * fPeriodLength * data.length;
+
+				boolean flatTrendConsistent = fTrendWidth != null && fTrendDegree != null &&
+						fTrendWidth == massiveWidth && fTrendDegree == 0;
```

Comment:
Does it make sense for `fTrendDegree` to be explicitly set? Would we want it to be anything other than 0 or 1?

Reply:
I give it the same flexibility that the original has. It could even be 2 in principle though I'm not sure if the original authors tested convergence in that case.